drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -730,7 +730,7 @@ static int nrf5_stop(const struct device *dev)
 	return 0;
 }
 
-#if defined(CONFIG_NRF_802154_CARRIER_FUNCTIONS)
+#if defined(CONFIG_IEEE802154_802154_CARRIER_FUNCTIONS)
```

Comment:
Should the `NRF_802154_CARRIER_FUNCTIONS` options `select IEEE802154_802154_CARRIER_FUNCTIONS`?

Reply:
I think `default y if (OPENTHREAD_DIAG || IEEE802154_CARRIER_FUNCTIONS)` would be better solution.